### PR TITLE
Attack pool

### LINKED ISSUE #40: Create AttackPool object
Needs to do things like

 - [x] What dice are in the pool
 - [x] Sum the damage
 - [x] Modify dice
 - [x] Other?

### PR DESCRIPTION
**Pull request by danbuis**
## Goal of Pull Request
Build an AttackPool object

## Potential Tasks

  - [x] Connects to #40
  - [x] Meets the DefDone of the linked issue
  - [x] All tests pass (Although the reviewer should run them to make sure)
  - [x] Other?

## Any other pertinent information
Uses a lot of enums to package non-dynamic information

### CODE REVIEW

File: src/gameComponents/AttackPool.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    private boolean spentBrace = false;
+    private boolean spentEvade = false;
+    private boolean spentContain = false;
+    private boolean spentScatter = false;
+    private boolean spentRedirect = false;
+    private int currentRolledDamage;
+    private ArrayList<Die> currentPool;
+    private boolean critEffect;
+
+
```
Comment (Sobonis01) [inline src/gameComponents/AttackPool.java:21]: //So the Boolean is to indicate the that a defense token was not in use, and impacting the attack token?//

File: src/gameComponents/AttackPool.java [hunk 2]
```diff
@@ -1,4 +1,32 @@
 package gameComponents;
 
 public class AttackPool {
+    private boolean spentBrace = false;
+    private boolean spentEvade = false;
+    private boolean spentContain = false;
+    private boolean spentScatter = false;
+    private boolean spentRedirect = false;
```
Comment (boredompwndu) [inline src/gameComponents/AttackPool.java]: don't forget salvo. Also thoughts for / against having an arraylist of spentTokens / tokentypes, and just seeing if future tokens are members of that list

File: src/gameComponents/DiceSymbol.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package gameComponents;
+
+public enum DiceSymbol {
+    CRIT, HIT, ACCURACY
```
Comment (boredompwndu) [inline src/gameComponents/DiceSymbol.java:4]: blanks?

**Review by Sobonis01 — DISMISSED** [review]

File: src/gameComponents/AttackPool.java (hunk 1 above)
Comment (danbuis) [reply src/gameComponents/AttackPool.java:21]: Basically a first draft to make sure we don't double brace an attack

File: src/gameComponents/DiceSymbol.java (hunk 3 above)
Comment (danbuis) [reply src/gameComponents/DiceSymbol.java:4]: Intentionally left out.  They aren't a symbol.

This is to count the occurances of a symbol to account for double icon faces.  Might have a side of yagni given the values stored in the enums.

File: src/gameComponents/AttackPool.java (hunk 2 above)
Comment (danbuis) [reply src/gameComponents/AttackPool.java]: Yeah, that's a better solution than a list of bools.

Comment (danbuis) [reply src/gameComponents/AttackPool.java]: Opted to strip this out entirely.  I'd prefer that this object really be about holding the current dice.  No need to have defense tokens mucking this up.  This block probably has a better home in the proposed Attack object that is serving as the go between for all things related to attacks, in this instance the go between for ships/tokens and the attack pool.

**Comment by danbuis** [comment]
Ok @boredompwndu I've made my edits.  Punting this one back over to you.

**Review by boredompwndu — DISMISSED** [review]

**Comment by boredompwndu** [comment]
branch has conflicts, just rule in favor of the develop branch, and things are fine

**Pull request merged**